lib/awful/titlebar.lua

```diff
@@ -26,6 +26,37 @@ local titlebar = {
     widget = {}
 }
 
+--- Set a declarative widget hierarchy description
+-- @param args An array containing the widgets disposition
+--
+-- Here is a simple example of a horizontal layout with 3 widgets, including one
+-- with a background and one that will only be displayed on screen 1
+--     mywibox:set_widgets {
+--         s == 1 and my_first_widget, -- Only display on screen 1
+--         my_second_widget,
+--         { -- Add a background color/pattern for my_third_widget
+--               my_third_widget,
+--               layout = wibox.widget.background(nil,beautiful.bg_focus),
+--         },
+--         layout = wibox.layout.fixed.horizontal,
+--     }
+--
+-- The `layout` array member can either be the name of the layout or a layout
+-- object itself. This allow additional parameters be passed to the layout. It
```

Comment:
Right, I also need to document the new features such as properties (see example above), new widget creation and widget identifiers.